msmbuilder/msm/validation/bootstrapmsm.py

```diff
@@ -0,0 +1,164 @@
+    matrices are obtained using random sampling with
+    replacement over the set of input trajectories.
+    The model also fits the mle over the original set.
+
+    Parameters
+    ----------
+    n_samples : int
+        Number of bootstrap models to construct
+    n_procs : int
+        Number of processors to use.
+        Defaults to int(cpu_count/2)
+    msm_args: dict
+        Dictionary containing arguments to pass unto
+        the MSM models.
+
+    Attributes
+    ----------
+    mle_ : Markov State Model
+        MSM model fit unto original dataset. The state mapping
+        inferred here is used through the rest of the models.
+    all_populations_ : list of lists
+        Array of all populations obtained from each model
+    resample_ind_ : list
+        lis of resample indices used to fit each bootstrap mdl.
+        This can be used to regenerate any mdl without having
+        to store it inside this object.
+    mapped_populations_: array (n_samples, mle.n_states_)
+        Array containing population estimates from all the
+        models for the states retained in the mle model.
+    mapped_populations_mean_: array shape = (mle.n_states_)
+        Mean population across the set of models for the states
+        contained in the mle model.
+    mapped_populations_std_: array shape = (mle.n_states_)
+        Population std across the set of models for the states
+        contained in the mle model.
+    mapped_populations_sem_: array shape = (mle.n_states_)
+         Population sem across the set of models for the states
+         contained in the mle model.(std/self._succesfully_fit)
+
+    Notes
+    -----
+    The correct number of bootstrap sample is subject to
+    debate with several hundred samples(n_samples)
+    being the recommended starting figure.
+
+    The fit function for this model optionally takes in a pool
+    of workers making it capable of parallelizing across
+    compute nodes via mpi or ipyparallel. This can lead to
+    a significant speed up for larger number of samples.
+    """
+    def __init__(self, n_samples=10,  n_procs=None, msm_args={}):
+        self.n_samples = n_samples
+        self.n_procs = n_procs
+        self.msm_args = msm_args
+        self.mle_ = MarkovStateModel(**self.msm_args)
+
+        self._succesfully_fit = 0
+        self.all_populations_ = None
+        self.mapped_populations_ = None
+        self.resample_ind_ = None
+
+    def fit(self, sequences, y=None, pool=None):
+        sequences = list_of_1d(sequences)
+        self.mle_.fit(sequences, y=y)
+        self._parallel_fit(sequences, pool)
+
+
+    def _fit_one(self, sequences):
+        mdl = MarkovStateModel(**self.msm_args)
+        #there is no guarantee that the mdl fits this sequence set so
+        #we return None in that instance.
+        try:
+            mdl.fit(sequences)
+            # solve the eigensystem
+            mdl.eigenvalues_[0]
+        except:
```

Comment:
ohh good point @mpharrigan . i was thinking of the value error that pops up when fitting msms every now and then. Gonna change it to explicitly expect the value error. .